src/sycl/Framework/RunningAverage.h

```diff
@@ -0,0 +1,53 @@
+#ifndef FWCore_Utilities_RunningAverage_H
+#define FWCore_Utilities_RunningAverage_H
+#include <atomic>
+#include <algorithm>
+#include <array>
+
+// Function for testing RunningAverage
+namespace test_average {
+  namespace running_average {
+    int test();
+  }
+}  // namespace test_average
+
+namespace edm {
+  // keeps the running average of the last N entries
+  // thread safe, fast: does not garantee precise update in case of collision
+  class RunningAverage {
+    // For tests
+    friend int ::test_average::running_average::test();
+
+  public:
+    static constexpr int N = 16;  // better be a power of 2
+    explicit RunningAverage(unsigned int k = 4) : m_mean(N * k), m_curr(0) {
+      for (auto& i : m_buffer)
+        i = k;
+    }
+
+    int mean() const { return m_mean / N; }
+
+    int upper() const {
+      auto lm = mean();
+      return lm += (sycl::abs(m_buffer[0] - lm) + sycl::abs(m_buffer[N / 2] - lm));
```

Comment:
With explicit use of `sycl::abs()` I'd suggest to add `#include <CL/sycl.hpp>` to make the header dependence more explicit.